monai/data/nifti_writer.py

```diff
@@ -9,33 +9,126 @@
+    multiple channels, time and/or modality axes should be appended after the
+    first three dimensions.
 
-def write_nifti(data, file_name, affine=None, target_affine=None, dtype=np.float32):
-    """Write numpy data into nifti files to disk.
 
     Args:
         data (numpy.ndarray): input data to write to file.
-        affine (numpy.ndarray): affine information for the data.
         file_name (string): expected file name that saved on disk.
-        target_affine (numpy.ndarray, optional):
-            before saving the (data, affine), transform the data into the orientation defined by `target_affine`.
+        affine (numpy.ndarray): the current affine of `data`. Defaults to `np.eye(4)`
+        target_affine (numpy.ndarray, optional): before saving
+            the (`data`, `affine`) as a Nifti1Image,
+            transform the data into the coordinates defined by `target_affine`.
+        resample (bool): whether to run resampling when the target affine
+            could not be achieved by swapping/flipping data axes.
+        output_shape (None or tuple of ints): output image shape.
+            this option is used when resample = True.
+        interp_order (int): the order of the spline interpolation, default is 3.
+            The order has to be in the range 0 - 5.
+            https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.affine_transform.html
+            this option is used when `resample = True`.
+        mode (`reflect|constant|nearest|mirror|wrap`):
+            The mode parameter determines how the input array is extended beyond its boundaries.
+            this option is used when `resample = True`.
+        cval (scalar): Value to fill past edges of input if mode is "constant". Default is 0.0.
+            this option is used when `resample = True`.
         dtype (np.dtype, optional): convert the image to save to this data type.
     """
     assert isinstance(data, np.ndarray), 'input data must be numpy array.'
+    sr = min(data.ndim, 3)
     if affine is None:
-        affine = np.eye(4)
+        affine = np.eye(4, dtype=np.float64)
+    affine = to_affine_nd(sr, affine)
 
     if target_affine is None:
-        results_img = nib.Nifti1Image(data.astype(dtype), affine)
-    else:
-        start_ornt = nib.orientations.io_orientation(affine)
-        target_ornt = nib.orientations.io_orientation(target_affine)
-        ornt_transform = nib.orientations.ornt_transform(start_ornt, target_ornt)
+        target_affine = affine
+    target_affine = to_affine_nd(sr, target_affine)
 
-        reverted_results = nib.orientations.apply_orientation(data, ornt_transform)
-        results_img = nib.Nifti1Image(reverted_results.astype(dtype), target_affine)
+    if np.allclose(affine, target_affine):
+        # no affine changes, save (data, affine)
+        results_img = nib.Nifti1Image(data.astype(dtype), to_affine_nd(3, target_affine))
```

Comment:
shouldn't be "to_affine_nd(sr, target_affine)"?

Reply:
no, Nifti1Image requires a 4x4 affine